src/main/java/com/google/cloud/teleport/splunk/SplunkEventWriter.java

```diff
@@ -291,6 +303,23 @@ private static void flushWriteFailures(
     }
   }
 
+  /**
+   * Checks whether the HEC URL matches the format PROTOCOL://HOST[:PORT].
+   *
+   * @param url for HEC event collector
+   * @return true if the URL is valid
+   */
+  private static boolean isValidUrlFormat(String url) {
+    Pattern pattern = Pattern.compile("^http(s?)://([^:]+)(:[0-9]+)?$");
+    Matcher matcher = pattern.matcher(url);
+    if (matcher.find()) {
+      String host = matcher.group(2);
+      return (InetAddresses.isInetAddress(host) || InternetDomainName.isValid(host))
+          && !host.endsWith(HttpEventPublisher.HEC_URL_PATH);
```

Comment:
This check is redundant.  Regex allows 2 types of urls: 
a) urls with port number (i.e protocol://hostname:port)
b) urls  without port (i.e protocol://hostname).  

It automatically returns false for any url that has path fragment  (ie protocol://hostname[:port]/) where trailing slash & anything beyond that is considered as path fragment. For e.g in url  ```https://nonporttrailingslash.splunk.com/ ``` the host will be nonporttrailingslash.splunk.com/  and InetAddresses.isInetAddress(host) returns false due to trailing slash.  Hence  ```https://nonport.splunk.com/services/collector/event``` will return false even without host.endsWith(HttpEventPublisher.HEC_URL_PATH) explicit check.

Reply:
Oh, I missed that! Thanks for the detailed explanation :) Fixed it now